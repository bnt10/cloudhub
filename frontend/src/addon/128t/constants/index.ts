import {
  ROUTER_TABLE_SIZING,
  TOPSOURCES_TABLE_SIZING,
  TOPSESSIONS_TABLE_SIZING,
} from 'src/addon/128t/constants/tableSizing'
import {GET_ALLROUTERS_INFO} from 'src/addon/128t/constants/query'
import {
  SALT_FULL_DIRECTORY,
  SALT_MIN_DIRECTORY,
} from 'src/addon/128t/constants/directory'

export {
  ROUTER_TABLE_SIZING,
  TOPSOURCES_TABLE_SIZING,
  TOPSESSIONS_TABLE_SIZING,
  GET_ALLROUTERS_INFO,
  SALT_FULL_DIRECTORY,
  SALT_MIN_DIRECTORY,
}
