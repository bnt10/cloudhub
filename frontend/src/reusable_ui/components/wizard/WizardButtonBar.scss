/*
  Wizard Button Bar Styles
  ------------------------------------------------------------------------------
*/

.wizard-button-bar {
  display: inline-flex;
  flex-shrink: 0;
  margin: $ix-marg-b auto $ix-marg-c auto;
  position: relative;
  min-width: 100%;
  justify-content: center;
  align-items: center;
  height: auto;
  flex-wrap: nowrap;
  .btn {
    position: relative;
    white-space: nowrap;
    margin: 0 $ix-marg-a;
  }
}
