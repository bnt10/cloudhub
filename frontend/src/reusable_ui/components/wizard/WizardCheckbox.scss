/*
  Wizard Checkbox Component
  ------------------------------------------------------------------------------
*/

.wizard-checkbox--group {
  margin-top: $ix-marg-c;
  color: $g11-sidewalk;
  transition: color 0.25s ease;
  font-weight: 500;

  .wizard-checkbox--label {
    margin-left: $ix-marg-b;
    font-size: 14px;
  }
  .wizard-checkbox--subtext{
    margin-left: $ix-marg-b+36;
    font-size: 11px;
  }
  &:hover {
    color: $g18-cloud;
    cursor: pointer;
  }
}