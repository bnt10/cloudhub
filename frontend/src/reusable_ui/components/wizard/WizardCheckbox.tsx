import React, {PureComponent} from 'react'
import {ErrorHandling} from 'src/shared/decorators/errors'
import SlideToggle from 'src/reusable_ui/components/slide_toggle/SlideToggle'
import {ComponentColor, ComponentSize} from 'src/reusable_ui/types'

interface Props {
  isChecked: boolean
  text: string
  subtext?: string
  onChange: (isChecked: boolean) => void
}

@ErrorHandling
class WizardCheckbox extends PureComponent<Props> {
  public render() {
    const {text, isChecked, subtext} = this.props

    return (
      <div className="form-group col-xs-12">
        <div className="form-control-static wizard-checkbox--group">
          <SlideToggle
            color={ComponentColor.Success}
            size={ComponentSize.ExtraSmall}
            active={isChecked}
            onChange={this.onChangeSlideToggle}
            tooltipText={text}
          />
          <span
            className="wizard-checkbox--label"
            onClick={this.onChangeSlideToggle}
          >
            {text}
          </span>
          {subtext && (
            <span className="wizard-checkbox--subtext">{subtext}</span>
          )}
        </div>
      </div>
    )
  }

  private onChangeSlideToggle = () => {
    const {onChange, isChecked} = this.props
    onChange(!isChecked)
  }
}

export default WizardCheckbox
