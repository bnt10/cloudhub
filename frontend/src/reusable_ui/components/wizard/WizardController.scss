/*
  Wizard Controller Styles
  ------------------------------------------------------------------------------
*/

.wizard-step-title {
  text-align: center;
  @include no-user-select();
}

.wizard-controller {
  display: inline-flex;
  flex-direction: column;
  min-width: 100%;
  align-items: center;
}

.wizard-skip-link {
  flex-shrink: 0;
}

.wizard-tip-text {
  min-width: 100%;
  color: $g12-forge;
  text-align: center;
  p {
    font-weight: 600;
  }
}
