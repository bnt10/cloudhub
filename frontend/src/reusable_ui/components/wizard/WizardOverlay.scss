/*
  Overlay Wizard Styles
  ------------------------------------------------------------------------------
*/

.wizard-overlay {
  .overlay--body {
    padding: 30px 0;
  }
}