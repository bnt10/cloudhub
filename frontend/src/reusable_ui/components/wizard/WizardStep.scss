/*
  Wizard Step Styles
  ------------------------------------------------------------------------------
*/

.wizard-step--container {
  position: relative;
  overflow: hidden;
  margin: $ix-marg-c 0;
  width: 100%;
}

.wizard-step--child {
  padding: $ix-marg-c;
}

.wizard-step--shadow {
  position: absolute;
  min-width: 100%;
  min-height: 100%;
  pointer-events: none;
  z-index: 100;
  box-shadow: inset 0px -11px 8px -10px $g0-obsidian,
    inset 0px 11px 8px -10px $g0-obsidian;
}
