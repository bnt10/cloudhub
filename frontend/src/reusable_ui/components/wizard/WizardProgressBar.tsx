// Libraries
import React, {PureComponent} from 'react'

// Components
import ProgressConnector from 'src/reusable_ui/components/wizard/ProgressConnector'

// Types
import {Step} from 'src/types/wizard'
import {ConnectorState, StepStatus} from 'src/reusable_ui/constants/wizard'

import {ErrorHandling} from 'src/shared/decorators/errors'

interface Props {
  steps: Step[]
  currentStepIndex: number
  handleJump: (jumpIndex: number) => () => void
}

@ErrorHandling
class WizardProgressBar extends PureComponent<Props> {
  public render() {
    return <div className="wizard-progress-bar">{this.WizardProgress}</div>
  }

  private get WizardProgress(): JSX.Element {
    const {steps, currentStepIndex, handleJump} = this.props
    const progressBar = steps.reduce((acc, step, index) => {
      const {stepStatus} = step
      let currentStep = ''

      const isCurrentStep = index === currentStepIndex
      const isNotErrored = stepStatus !== StepStatus.Error
      const isAtEndOfWizard = currentStepIndex === steps.length - 1
      const isCompleted = stepStatus === StepStatus.Complete
      const isLastStep = index === steps.length - 1
      const previousIsComplete =
        steps[index - 1] && steps[index - 1].stepStatus === StepStatus.Complete

      // STEP STATUS
      if (isCurrentStep && isNotErrored) {
        currentStep = 'circle-thick current'
      }

      if (isLastStep && previousIsComplete) {
        currentStep = 'checkmark'
      }

      const stepEle = (
        <div
          key={`stepEle${index}`}
          className="wizard-progress-button"
          onClick={handleJump(index)}
        >
          <div className="wizard-progress-title">{step.title}</div>
          <span className={`icon ${currentStep || stepStatus}`} />
        </div>
      )

      // PROGRESS BAR CONNECTOR
      let connectorStatus = ConnectorState.None

      if (isCurrentStep && isNotErrored) {
        connectorStatus = ConnectorState.Some
      }

      if (isAtEndOfWizard || isLastStep || isCompleted) {
        connectorStatus = ConnectorState.Full
      }

      const connectorEle = isLastStep ? null : (
        <ProgressConnector
          key={`connectorEle${index}`}
          status={connectorStatus}
        />
      )

      return [...acc, stepEle, connectorEle]
    }, [])
    return <>{progressBar}</>
  }
}

export default WizardProgressBar
