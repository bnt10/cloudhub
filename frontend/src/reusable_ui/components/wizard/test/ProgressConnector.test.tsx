import React from 'react'
import {shallow} from 'enzyme'

import ProgressConnector from 'src/reusable_ui/components/wizard/ProgressConnector'
import {ConnectorState} from 'src/reusable_ui/constants/wizard'

describe('Progress Connector', () => {
  let wrapper

  const expectedWithoutStatus = `wizard-progress-connector wizard-progress-connector--${
    ConnectorState.None
  }`

  const expectedWithStatusSome = `wizard-progress-connector wizard-progress-connector--${
    ConnectorState.Some
  }`

  const expectedWithStatusAll = `wizard-progress-connector wizard-progress-connector--${
    ConnectorState.Full
  }`

  describe('without Props', () => {
    const props = {
      status: undefined,
    }

    beforeEach(() => (wrapper = shallow(<ProgressConnector {...props} />)))

    it('mounts without exploding', () => {
      expect(wrapper).toHaveLength(1)
    })

    it('defaults to enum state "none" when no props provided', () => {
      expect(wrapper.find('span').props().className).toBe(expectedWithoutStatus)
    })

    it('matches snapshot when provided minimal props', () => {
      expect(wrapper).toMatchSnapshot()
    })
  })

  describe('with status: some', () => {
    const props = {
      status: ConnectorState.Some,
    }

    beforeEach(() => (wrapper = shallow(<ProgressConnector {...props} />)))

    it('defaults to enum state "some" when no props provided', () => {
      expect(wrapper.find('span').props().className).toBe(
        expectedWithStatusSome
      )
    })

    it('matches snapshot when provided "some" status', () => {
      expect(wrapper).toMatchSnapshot()
    })
  })

  describe('with status: all', () => {
    const props = {
      status: ConnectorState.Full,
    }

    beforeEach(() => (wrapper = shallow(<ProgressConnector {...props} />)))

    it('defaults to enum state "all" when no props provided', () => {
      expect(wrapper.find('span').props().className).toBe(expectedWithStatusAll)
    })

    it('matches snapshot when provided "full" status', () => {
      expect(wrapper).toMatchSnapshot()
    })
  })
})
