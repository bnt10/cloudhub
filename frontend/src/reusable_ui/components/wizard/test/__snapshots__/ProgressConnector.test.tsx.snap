// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Progress Connector with status: all matches snapshot when provided "full" status 1`] = `
<span
  className="wizard-progress-connector wizard-progress-connector--full"
/>
`;

exports[`Progress Connector with status: some matches snapshot when provided "some" status 1`] = `
<span
  className="wizard-progress-connector wizard-progress-connector--some"
/>
`;

exports[`Progress Connector without Props matches snapshot when provided minimal props 1`] = `
<span
  className="wizard-progress-connector wizard-progress-connector--none"
/>
`;
